modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/SelectTracksWorkflowOperationHandler.java

```
@@ -0,0 +1,500 @@
+        final AugmentedTrack singleAudioTrack = singleAudioTrackOpt.get();
+        final String forceTargetOpt = getConfiguration(workflowInstance, CONFIG_FORCE_TARGET)
+                .orElse(FORCE_TARGET_DEFAULT);
+
+        final Optional<AugmentedTrack> forceTargetTrackOpt = findTrackByFlavorType(augmentedTracks, forceTargetOpt);
+
+        if (!forceTargetTrackOpt.isPresent()) {
+          throw new IllegalStateException(
+                  String.format("\"%s\" set to \"%s\", but target flavor \"%s\" not found!",
+                          CONFIG_AUDIO_MUXING,
+                          AudioMuxing.FORCE.value, forceTargetOpt));
+        }
+
+        final AugmentedTrack forceTargetTrack = forceTargetTrackOpt.get();
+
+        if (singleAudioTrack.track != forceTargetTrack.track) {
+          // Copy it over...
+          final TrackJobResult muxResult = mux(forceTargetTrack.track, singleAudioTrack.track, mediaPackage);
+          result.add(muxResult);
+
+          // ...and remove the original
+          final TrackJobResult hideAudioResult = hideAudio(singleAudioTrack.track, mediaPackage);
+          result.add(hideAudioResult);
+        }
+      } else {
+        // No special options selected, or conditions for special options don't match.
+        final MuxResult muxResult = muxMultipleVideoTracks(mediaPackage, augmentedTracks);
+        result.add(muxResult);
+      }
+    } else {
+      // Second case: we have exactly one video track that is not hidden (hopefully, because for all other cases there
+      // were no requirements given).
+      final MuxResult muxResult = muxSingleVideoTrack(mediaPackage, augmentedTracks);
+      result.add(muxResult);
+    }
+
+    final MediaPackageElementFlavor targetTrackFlavor = MediaPackageElementFlavor.parseFlavor(StringUtils.trimToNull(
+            getConfiguration(workflowInstance, "target-flavor")
+                    .orElseThrow(() -> new IllegalStateException("Target flavor not specified"))));
```

Fail fast. No need to handle all the media file operations if we finally fail with a configuration error anyway.